testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2673,12 +2673,10 @@ protected void assertCreateTableAsSelect(Session session, @Language("SQL") Strin
     @Test
     public void testCreateTableAsSelectNegativeDate()
     {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+
         String tableName = "negative_date_" + randomNameSuffix();
 
-        if (!hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA)) {
-            assertQueryFails(format("CREATE TABLE %s AS SELECT DATE '-0001-01-01' AS dt", tableName), "This connector does not support creating tables with data");
-            return;
-        }
```

Comment:
this validation seemed ok. For connectors which say they don't support `CREATE_TABLE_WITH_DATA` we just verify that they are not lying.

It does look redundant since `testCreateTableAsSelect` already verifies that though.